tests/integration/test_storage_nats/test_nats_jet_stream.py

```diff
@@ -0,0 +1,1766 @@
+
+    while time.monotonic() < deadline:
+        result = instance.query("SELECT count() FROM test.view_many")
+        logging.debug(result, messages_num * threads_num)
+        if int(result) >= messages_num * threads_num:
+            break
+        time.sleep(1)
+
+    assert (
+        int(result) == messages_num * threads_num
+    ), "ClickHouse lost some messages or got duplicated ones. Total count: {}".format(
+        result
+    )
+
+
+def test_nats_overloaded_insert(nats_cluster):
+
+    asyncio.run(add_durable_consumer(cluster, "test_stream", "test_consumer"))
+
+    instance.query(
+        f"""
+        CREATE TABLE test.nats_consume (key UInt64, value UInt64)
+            ENGINE = NATS
+            SETTINGS nats_url = 'nats1:4444',
+                     nats_stream = 'test_stream',
+                     nats_consumer_name = 'test_consumer',
+                     nats_subjects = 'test_subject',
+                     nats_num_consumers = 5,
+                     nats_max_block_size = 10000,
+                     nats_format = 'TSV',
+                     nats_row_delimiter = '\\n';
+        CREATE TABLE test.nats_overload (key UInt64, value UInt64)
+            ENGINE = NATS
+            SETTINGS nats_url = 'nats1:4444',
+                     nats_stream = 'test_stream',
+                     nats_subjects = 'test_subject',
+                     nats_format = 'TSV',
+                     nats_row_delimiter = '\\n';
+        CREATE TABLE test.view_overload (key UInt64, value UInt64)
+            ENGINE = MergeTree
+            ORDER BY key
+            SETTINGS old_parts_lifetime=5, cleanup_delay_period=2, cleanup_delay_period_random_add=3,
+            cleanup_thread_preferred_points_per_iteration=0;
+        CREATE MATERIALIZED VIEW test.consumer_overload TO test.view_overload AS
+            SELECT * FROM test.nats_consume;
+        """
+    )
+
+    messages_num = 100000
+
+    def insert():
+        values = []
+        for i in range(messages_num):
+            values.append("({i}, {i})".format(i=i))
+        values = ",".join(values)
+
+        while True:
+            try:
+                instance.query(
+                    "INSERT INTO test.nats_overload VALUES {}".format(values)
+                )
+                break
+            except QueryRuntimeException as e:
+                if "Local: Timed out." in str(e):
+                    continue
+                else:
+                    raise
+
+    threads = []
+    threads_num = 5
+    for _ in range(threads_num):
+        threads.append(threading.Thread(target=insert))
+    for thread in threads:
+        time.sleep(random.uniform(0, 1))
+        thread.start()
+
+    time_limit_sec = 300
+    deadline = time.monotonic() + time_limit_sec
+
+    while time.monotonic() < deadline:
+        result = instance.query("SELECT count() FROM test.view_overload")
+        time.sleep(1)
+        if int(result) >= messages_num * threads_num:
+            break
```

Comment:
Could you please either use `query_with_retry` (which kind of ignores exceptions) or create a function for such loops? I think it would make the tests more readable and shorter. I won't comment on this issue anymore, but could you please check all the occurences?